Dynatrace/dynatrace-operator
# Refactors the `dtclient_reconciler`

# Description

Refactors the `dtclient_reconciler` to simplify and abstract logic for checking tokens and creating an api client.

## How can this be tested?

1. Create a cluster and deploy the operator
2. Make sure nothing crashed so far
3. Apply a dynakube __without__ a token secret 
  * Check that the operator prints the appropriate log that it is indeed missing
  * Check that the description of the dynakube has the appropriate condition set
4. Repeat step 3. for the following scenarios:
  * A token secret exists but it is empty
  * A token secret exists but the values are invalid (have whitespace at the beginning and the end of the token value)
  * A token secret exists with a valid token, but the token is missing scopes
  * A token secret exists with a valid token that has all necessary scopes
5. Run unit tests
6. Run E2E tests

## Checklist
